pre/remap_restart/remap_command_line.py

```diff
@@ -0,0 +1,171 @@
+import shlex
+import ruamel.yaml
+import shutil
+import questionary
+import glob
+import argparse
+from remap_utils import *
+from remap_params import * 
+
+def parse_args(program_description):
+
+    parser = argparse.ArgumentParser(description='Remap restarts',epilog=program_description,formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    p_sub = parser.add_subparsers(help = 'Sub command help')
+    p_config  = p_sub.add_parser(
+     'config_file',
+     help = "Use config file as input",
+    )
+
+    p_config.add_argument('-c', '--config_file',  help='YAML config file')
+
+    p_command = p_sub.add_parser(
+     'command_line',
+     help = "Use command line as input",
+    )
+    p_command.add_argument('-merra2', action='store_true', default= False, help='use merra2 restarts')
+    p_command.add_argument('-ymdh',              help='yyyymmddhh year month date hour of input and new restarts')
+    p_command.add_argument('-grout',             help='Grid ID of new restarts, format Cxx')
+    p_command.add_argument('-levsout',           help='Levels for new restarts')
+
+    p_command.add_argument('-out_dir',           help='Directory for new restarts')
+    p_command.add_argument('-rst_dir',           help='Directory of input restarts')
+
+    p_command.add_argument('-expid',             help='Experiment ID of input restarts')
+    p_command.add_argument('-newid', default="", help='Experiment ID for new restarts')
+
+    p_command.add_argument('-bcvin',             help='Boundary conditions version of input restarts')
+    p_command.add_argument('-bcvout',            help='Boundary conditions version for new restarts')
+
+    p_command.add_argument('-in_wemin',          help='Min. water snow water equivalent param. used with input restarts')
+    p_command.add_argument('-out_wemin',         help='Min. water snow water equivalent param. to be used with new restarts')
+
+    ocean_grids=['360x180','1440x720','2880x1440','CS', '72x36', '360x200','720x410','1440x1080']
+
+    p_command.add_argument('-oceanin',           help='Ocean horizontal grid of input restarts. \n \
+                                                       data model choices: 360x180,1440x720,2880x1440,CS. \n \
+                                                       coupled model choices: 72x36, 360x200,720x410,1440x1080', choices=ocean_grids)
+    p_command.add_argument('-oceanout',          help='Ocean horizontal grid of new restarts. \n \
+                                                       choices are the same as option "oceanin"', choices=ocean_grids)
+
+    p_command.add_argument('-ocnmdlin',   default='data',     help='Ocean model of input restarts',  choices=['data', 'MOM5', 'MOM6'])
+    p_command.add_argument('-ocnmdlout',  default='data',     help='Ocean model for new restarts', choices=['data', 'MOM5', 'MOM6'])
+    p_command.add_argument('-catch_model',default='catch',    help='Catchment[CN] model', choices=['catch', 'catchcnclm40', 'catchcnclm45'])
```

Comment:
@weiyuan-jiang : Is this needed?  It's not in the "remap_questionary.py", so perhaps it's now deduced from the restart file name?

Reply:
I think we have to keep it here for command_line.  To make it simple, there is no deduction for command_line options